articles/migrations/guides/legacy-lock-api-deprecation.md

```diff
@@ -0,0 +1,195 @@
+* Invoking the `getSSOData()` function will now trigger a call to the [/authorize](/api/authentication#authorize-client) endpoint, which will in turn result in the execution of [rules](/rules).
+
+##### Polling for an existing session
+
+In some multi-application scenarios, where a user logging out of one application needs to be logged out of other applications, an application may have been set up to periodically poll Auth0 using `getSSOData()` to see if a session existed, and if not, log the user out of the application. 
+
+Instead of doing this, applications should now use `checkSession()` instead of `getSSOData()`. The `getSSOData()` function performs more work behind the scenes than is needed for this purpose and applications that are not switched to `checkSession()` will suffer a needless performance penalty.
+
+The poll interval between checks to `checkSession()` should be at least 10 seconds between calls to avoid any issues in the future with rate limiting of this call.
+
+#### Web applications
+
+In "web applications", the backend typically has a session for the user. Over time, the application session may expire, in which case the application should renew the session. The application backend should invoke a call to the [/authorize](/api/authentication#authorize-client) endpoint to get a new token. If the Authorization Server (Auth0 in this case) still has a session for the user, the user will not have to re-enter their credentials to log in again. If Auth0 no longer has a session for the user, the user has to log in again.
+
+Customers with web applications who make server-side calls as part of the [Authorization Code Grant Flow](/api-auth/grant/authorization-code), and use the `response_type = code`, should use this approach. Specifically, such applications should make an /authorize request to refresh the token, which effectively renews the application session.
+
+### How to log users out
+
+The deprecation does not require any changes for [logout](/logout), but if a custom domain has been configured and is used when invoking authentication, the /logout endpoint should be invoked using the custom domain as well.
+
+## Troubleshooting
+
+### How to tell if you have deprecated usage
+
+Please take a look at the [Deprecation Error Reference](/articles/errors/deprecation-errors) to assist with verifying that your application does, or does not, use deprecated features.
+
+### How to test whether you are ready before the removal of service date
+
+Auth0 has provided a toggle in the tenant settings in the [Dashboard](${manage_url}) to allow customers to simulate what will stop working on the deprecation deadline of July 16, 2018. Navigate to the tenant settings screen, **Advanced** tab and scroll down to the block of migration toggles.
+
```

Comment:
Is not to 'simulate', right? It's to force the endpoint to stop working so you can make sure your applications do not depend on it.

Reply:
+1 to the above. I'd say something like
> Auth0 has provided a toggle in the tenant settings in the [Dashboard](${manage_url}) to allow customers to turn off the legacy endpoints manually, ahead of the deprecation deadline of July 16th, 2018. Navigate [...].